src/commands.cpp

```diff
@@ -8182,4 +8182,130 @@ void CommandFolderLinkInfo::procresult()
     }
 }
 
+CommandSyncPut::CommandSyncPut(MegaClient *client, SyncType type, handle nodeHandle, const string& localFolder, const std::string &deviceId, const string& syncName, int state, int subState, const string& extraData)
+{
+    assert(type != SyncType::INVALID);
+
+    cmd("sp");
+
+    arg("t", type);
+    arg("h", (byte*)&nodeHandle, MegaClient::NODEHANDLE);
+    arg("l", localFolder.c_str());
+    arg("d", deviceId.c_str());
+    arg("n", syncName.c_str());
+    arg("s", state);
+    arg("ss", subState);
+    arg("e", extraData.c_str());
+
+    tag = client->reqtag;
+}
+
+CommandSyncPut::CommandSyncPut(MegaClient* client, handle syncId, SyncType type, handle nodeHandle, const char* localFolder, const char *deviceId, const char* syncName, int state, int subState, const char* extraData)
+{
+    cmd("sp");
+
+    arg("id", syncId);
+
+    if (type != SyncType::INVALID)
+    {
+        arg("t", type);
+    }
+
+    if (nodeHandle != UNDEF)
+    {
+        arg("h", (byte*)&nodeHandle, MegaClient::NODEHANDLE);
+    }
+
+    if (localFolder)
+    {
+        arg("l", localFolder);
+    }
+
+    if (deviceId)
+    {
+        arg("d", deviceId);
+    }
+
+    if (syncName)
+    {
+        arg("n", syncName);
+    }
+
+    if (state > 0)
+    {
+        arg("s", state);
+    }
+
+    if (subState > 0)
+    {
+        arg("ss", subState);
+    }
+
+    if (extraData)
+    {
+        arg("e", extraData);
+    }
+
+    tag = client->reqtag;
+}
+
+void CommandSyncPut::procresult()
+{
+    Error e;
+    if (checkError(e, client->json))
+    {
+        return client->app->syncput_result(e, UNDEF);
+    }
+
+    handle h = client->json.gethandle(MegaClient::NODEHANDLE); // [sync id]
```

Comment:
picky suggestion: rename h to syncID and remove the comment

Reply:
Yes, it's ok for me. I'll modify it